### DISCUSSION TITLE
Deferred Column Loading with Optional raiseload

### DISCUSSION THREAD

**Discussion by cancan101** [sqlalchemy/sqlalchemy#7038] [Usage Questions]
I would like to specify a set of columns on a model to be `deferred` and then at query time decide if I want to `raiseload` or not if those columns are accessed without being undeferred. An example of what that would look like is:

```
python
class Book(Base):
    ....
    photo = deferred(Column(Binary))

...
raiseload=True
query = session.query(Book).options(
     defer("*", raiseload=raiseload),
)
```
where the `*` here means raise load for any deferred column that is accessed. At the query construction, the user would not have to enumerate all deferred columns in the model.

Is there any way to do that?

**Comment by zzzeek** [comment]
i would assume trying that didnt work ?   `defer("*", raiseload=True)`  as you have should set raiseload for all unloaded columns, it's just this includes columns that are expired by the session also, if you are using objects with expiration (edit: this answer is dumb, see reply)

**Reply by zzzeek** [reply]
OK I guess that applies to every column at loading time.  I think you instead need to set these columns up for "deferred(raiseload=True)" by default on your model.  that way you can undefer the ones you need at query time.

**Reply by zzzeek** [reply]
what's nice about undefer is that you can also use undefer_group() to undefer a set of columns under a named group at once.

**Reply by cancan101** [reply]
I do like to decouple the model definition from the loading (or erroring) strategy (query creation). 

I would be fine setting all deferred columns to raise at model definition and then marking them as "no raise" at query time, but there isn't away to wildcard change the defer behavior without actually undeferring, is there?

**Reply by zzzeek** [reply]
you can use the wildcard but it applies to all columns not just the deferred one.

what's implied here is that since raiseload was added as a flag to defer(), we should add a defer_group() option that works like undefer_group().  that's the way to "undefer" a set of columns that are part of a named "defer" category only without naming them.  it never made sense for there to be a "defer_group()" since the cols are already deferred, but now that there's a raiseload modifier, it actually does make sense.   i can support that being added.
